doc/manual/src/architecture/store/reference-scanning.md

```diff
@@ -0,0 +1,52 @@
+# Reference scanning
+
+In most cases, Nix can automatically come up with the references that store objects ought to have, saving the user from doing additional bookkeeping.
+
+## Why it works
+
+This might be surprising to users coming from elsewhere!
+Traditional build systems do not track dependencies between build artifacts, as the problem appears very challenging since file formats are numerous.
+It would seem either there would need to be a way to teach the build system how to parse arbitrary formats, or the build system would need to track references via manual annotations the user must write and, worse, keep up to date.
+I
+On of the best insights in the creation of Nix was how cut this Gordian knot.
+
+Store objects are, as previously mentioned, mounted at `<store-dir>/<hash>-<name>` when they are exposed to the file system.
+This is the textual form of a [store path](./paths.md).
```

Comment:
Isn't the store path the textual form of a store object reference?

Reply:
In `paths.md` there is a distinction between the conceptual pair of a 20 byte digest and name and the "string representation".